apps/nextjs/tests-e2e/tests/aila-chat/full-romans.test.ts

```diff
@@ -0,0 +1,111 @@
+import { clerkSetup, setupClerkTestingToken } from "@clerk/testing/playwright";
+import { test, expect, Page } from "@playwright/test";
+
+import { TEST_BASE_URL } from "../../config/config";
+import { bypassVercelProtection } from "../../helpers";
+import { continueChat, isFinished, waitForGeneration } from "./helpers";
+
+type FixtureMode = "record" | "replay";
+const FIXTURE_MODE = "replay" as FixtureMode;
+
+// TODO: enable shorter timeouts for replays when they include moderations
+const generationTimeout = FIXTURE_MODE === "record" ? 75000 : 50000;
+
+const applyFixtures = async (page: Page) => {
+  let fixtureName: string;
+
+  await page.route("**/api/chat", async (route) => {
+    const headers = route.request().headers();
+    headers["x-e2e-fixture-name"] = fixtureName;
+    headers["x-e2e-fixture-mode"] = FIXTURE_MODE;
+    await route.continue({ headers });
+  });
+
+  return {
+    setFixture: async (name: string) => {
+      fixtureName = name;
+    },
+  };
+};
+
+test.describe("Authenticated", { tag: "@authenticated" }, () => {
+  test("navigate to /aila as a signed-in user", async ({ page }) => {
+    await clerkSetup();
+    await bypassVercelProtection(page);
+    await setupClerkTestingToken({ page });
+
+    await page.goto(`${TEST_BASE_URL}/aila`);
+
+    const h1 = page.getByTestId("chat-h1");
+    await expect(h1).toBeInViewport();
+  });
+
+  test(
+    "Full aila flow with Romans fixture",
+    { tag: "@openai" },
+    async ({ page }) => {
+      test.setTimeout(generationTimeout * 5);
+
+      await test.step("Setup", async () => {
+        await clerkSetup();
+        await bypassVercelProtection(page);
+        await setupClerkTestingToken({ page });
+
+        await page.goto(`${TEST_BASE_URL}/aila`);
+        await expect(page.getByTestId("chat-h1")).toBeInViewport();
+      });
+
+      const { setFixture } = await applyFixtures(page);
+
+      await test.step("Fill in the chat box", async () => {
+        const textbox = page.getByTestId("chat-input");
+        const sendMessage = page.getByTestId("send-message");
+        const message = "Create a KS1 lesson on the end of Roman Britain";
+        await textbox.fill(message);
+        await expect(textbox).toContainText(message);
+
+        // TODO: the demo status doesn't seem to have been loaded yet so a demo modal is shown
+        await page.waitForTimeout(500);
```

Comment:
I think this will need a tweak in the demo code, but that's out of scope for right now